lib/pf/vlan/pool.pm

```diff
@@ -0,0 +1,154 @@
+package pf::vlan::pool;
+
+=head1 NAME
+
+pf::vlan::pool - Object oriented module for VLAN isolation oriented functions
+
+=head1 SYNOPSIS
+
+The pf::vlan::pool module contains the functions necessary for the VLAN Pool.
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+
+use pf::config;
+use pf::util;
+use pf::log();
+
+use pf::node;
+
+use Number::Range;
+
+=head1 SUBROUTINES
+
+Warning: The list of subroutine is incomplete
+
+=cut
+
+=head2 new
+
+Constructor.
+
+=cut
+
+sub new {
+    my $logger =  pf::log::get_logger();
+    $logger->debug("instantiating new pf::vlan::pool object");
+    my ( $class, %argv ) = @_;
+    my $this = bless {}, $class;
+    return $this;
+}
+
+sub getVlanPool {
+    my ($self, $vlan, $switch, $ifIndex, $mac, $node_info, $connection_type, $user_name, $ssid, $radius_request, $role) = @_;
+    my $logger =  pf::log::get_logger();
+
+    return $vlan if $self->rangeValidator($vlan);
+    my $range = Number::Range->new($vlan);
+
```

Comment:
Not sure I understand this part.
If the rangeValidator returns 1 we will never hit the getRoundRobin() ?